app.rb

```diff
@@ -0,0 +1,174 @@
+require_relative 'person'
+require_relative 'classroom'
+require_relative 'book'
+require_relative 'rental'
+require_relative 'student'
+require_relative 'teacher'
+
+class App
+  attr_reader :people, :books, :rentals
+
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_books
+    puts 'Listing all books:'
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_people
+    puts 'Listing all people:'
+    @people.each do |person|
+      person_type = person.is_a?(Student) ? 'Student' : 'Teacher'
+      puts "[#{person_type}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def list_people_with_numbers
+    puts 'Listing all people:'
+    @people.each_with_index do |person, index|
+      person_type = person.is_a?(Student) ? 'Student' : 'Teacher'
+      puts "#{index + 1}. #{person_type}: #{person.name}"
+    end
+  end
+
+  def create_person(type, name = nil, age = nil, parent_permission = nil)
+    case type
+    when 'student'
+      name = prompt_name(name)
+      age = prompt_age(age)
+      parent_permission = prompt_parent_permission(parent_permission)
+
+      @people << Student.new(age, name: name, parent_permission: parent_permission)
+    when 'teacher'
```

Comment:
- [ ] Kindly fix the issue while trying to create a student as the previous reviewer mentioned in his comments
- [ ] Add the changes requested by the previous reviewer to make the app works as expected 🥇

Reply:
@BregornOriginal, I updated line 47 as mentioned by the previous reviewer, and all his comments and then I fixed the linters. kindly check commits above.